src/modules/dpos/validator/hooks/useDelegateKey.js

```diff
@@ -0,0 +1,45 @@
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { regex } from 'src/const/regex';
+
+const generatorPublicKeyValidator = value => regex.generatorPublicKeyValidator.test(value);
+const blsPublicKeyValidator = value => regex.blsPublicKeyValidator.test(value);
+const proofOfPossessionValidator = value => regex.proofOfPossessionValidator.test(value);
+
+const useDelegateKeys = (initGeneratorKey, initBlsKey, initPop) => {
+  const [generatorPublicKey, setGeneratorPublicKey] = useState({
+    value: initGeneratorKey ?? '',
+    error: false,
+    message: '',
+  });
+  const [blsPublicKey, setBlsPublicKey] = useState({
+    value: initBlsKey ?? '',
+    error: false,
+    message: '',
+  });
+  const [proofOfPossession, setProofOfPossession] = useState({
+    value: initPop ?? '',
+    error: false,
+    message: '',
+  });
+  const { t } = useTranslation();
+
+  const setters = {
```

Comment:
if you need a hooks for validation and keep the state its need to reusable
```javascript
const useValidation = (initValue, regex, errorMessage) => {
 const [value, setValueState] = useState({
    value: initGeneratorKey ?? '',
    error: false,
    message: '',
  });
  
const setValue = (value) => {
   const isValid = regex.test(value);
   setValueState({
      value,
      error: !isValid,
      message: isValid ? '' : errorMessage,
    })
  };
  return [value, setValue];
  };
  ```
  usage
  ```javascript
  const message = t('Please enter a valid value')
const [BLSKey, setBLSKey] = useValidation('', regex.blsPublicKeyValidator, message)
const [pubKey, setPubKey] = useValidation('', regex.generatorPublicKeyValidator, message)
const [proofOfPossession, setProofOfPossession] = useValidation('', regex.proofOfPossessionValidator, message)
  ```
is it more readable?